src/components/CodeAndPreview.tsx

```diff
@@ -62,20 +73,38 @@ export function CodeAndPreview({ snippet }: Props) {
     }
   }, [Boolean(snippet)])
 
-  const { toast } = useToast()
-
+  // Update manual edits when snippet changes, with validation
   useEffect(() => {
     if (snippet?.manual_edits_json) {
-      setManualEditsFileContent(snippet.manual_edits_json)
+      try {
+        JSON.parse(snippet.manual_edits_json)
+        setManualEditsFileContent(snippet.manual_edits_json)
+      } catch (e) {
+        console.warn("Invalid manual edits JSON from snippet")
+        toast({
+          title: "Warning",
+          description:
+            "Invalid manual edits format in snippet. Using default template.",
+          variant: "destructive",
+        })
+        setManualEditsFileContent(JSON.stringify(manualEditsTemplate, null, 2))
+      }
     }
   }, [Boolean(snippet?.manual_edits_json)])
 
-  const userImports = useMemo(
-    () => ({
-      "./manual-edits.json": JSON.parse(manualEditsFileContent),
-    }),
-    [manualEditsFileContent],
-  )
+  // Safely parse userImports with error handling
+  const userImports = useMemo(() => {
+    try {
+      return {
+        "./manual-edits.json": JSON.parse(manualEditsFileContent),
+      }
+    } catch (e) {
+      console.warn("Error parsing manual edits for imports, using empty object")
+      return {
+        "./manual-edits.json": {},
+      }
+    }
+  }, [manualEditsFileContent])
```

Comment:
this try catch pattern is not good, let's revert that

Reply:
actually this is ok, but this file is getting too big so create a utility function in the utils directory `parseJsonOrDefault(manualEditsFileContent, {})`